packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -282,48 +372,67 @@ class InputDateTimePicker extends React.Component {
 		});
 	}
 
-	updateDateTime(date, time, textInput = getTextDate(date, time), errorMsg) {
-		const fullDate = (() => {
-			if (date === undefined || time === undefined) {
-				return undefined;
-			}
+	updateDatePartStateAndTriggerChange(event, dateStatePart) {
+		return new Promise(resolve => {
+			const lastInfos = {
+				datetime: this.state.datetime,
+				errorMessage: this.state.errorMessage,
+			};
+			this.setState(dateStatePart, () => {
+				const newInfos = {
+					datetime: dateStatePart.datetime,
+					errorMessage: dateStatePart.errorMessage,
+				};
+				this.triggerChange(event, lastInfos, newInfos);
+				resolve();
+			});
+		});
+	}
 
-			return setMinutes(date, time);
-		})();
+	triggerChange(event, lastInfos, newInfos = {}) {
+		const datetimeUpdated =
+			newInfos.datetime !== lastInfos.datetime &&
+			!isSameMinute(newInfos.datetime, lastInfos.datetime);
 
-		const fullDateUpdated =
-			fullDate !== this.state.lastFullDate && !isSameMinute(fullDate, this.state.lastFullDate);
+		const errorUpdated = newInfos.errorMessage !== lastInfos.errorMessage;
 
-		if (fullDateUpdated && this.props.onChange) {
-			this.props.onChange(fullDate);
+		if (this.props.onChange && (datetimeUpdated || errorUpdated)) {
+			this.props.onChange(event, newInfos.errorMessage, newInfos.datetime);
 		}
+	}
 
-		const errorUpdated = errorMsg !== this.state.lastErrMsg;
-		if (errorUpdated && this.props.onError) {
-			this.props.onError(errorMsg);
+	triggerBlur(event) {
+		if (this.props.onBlur) {
+			this.props.onBlur(event);
 		}
-
-		this.setState({
-			date,
-			time,
-			textInput,
-			lastFullDate: fullDate,
-			lastErrMsg: errorMsg,
-		});
 	}
 
 	render() {
 		const inputProps = omit(this.props, PROPS_TO_OMIT_FOR_INPUT);
+
+		const isDatetimeValid = isDateValid(this.state.datetime);
+		const inputFocused = this.state.inputFocused;
+		const isInternalInvalidDate = this.state.datetime === INTERNAL_INVALID_DATE;
+		const needInvalidPlaceholder = !isDatetimeValid && !isInternalInvalidDate && !inputFocused;
```

Comment:
`isDatetimeValid ` covers `isInternalInvalidDate `

Reply:
it's not exactly the same.
If `isDatetimeValid=true` then `isInternalInvalidDate=false`
but if `isDatetimeValid=false` then `isInternalInvalidDate` can be true or false

Reply:
We want to check if we need the invalid placeholder, si if the date is not valid, we will need it. No need to check if it's the internalInvalidDate that is already in the `isDatetimeValid` check.